lib/roby/app.rb

```diff
@@ -2997,11 +2997,18 @@ def remove_notification_listener(listener)
         # @param [Boolean] only_self if set, list only test files from within
         #   {#app_dir}. Otherwise, consider test files from all over {#search_path}
         # @return [Array<String>]
-        def discover_test_files(all: true, only_self: false)
+        def discover_test_files(all: true, only_self: false, base_dir: File.join(app_dir, "test"))
             if all
-                test_files = each_test_file_in_app.each_with_object({}) do |k, h|
+                robot_test_files = each_test_file_in_robot.each_with_object({}) do |k, h|
                     h[k] = []
                 end
+                base_dir_test_files = each_test_file_in(base_dir)
+                base_dir_test_files = base_dir_test_files.each(base_dir).with_object({}) do |k, h|
```

Comment:
@doudou is there a better way for doing that?

Reply:
See the enum_for comment below.

The whole transformation can also be done with `Hash[*each_test_file_in(base_dir).map { |k| [k, []] }]`. But I'd prefer the each_with_object solution personally.